os/include/tinyara/mpu.h

```diff
@@ -47,9 +47,18 @@ enum {
 #define MPU_NUM_REGIONS		1
 #endif
 
+/* Number of reserved MPU regions for a board. This is defined by the vendor.
+ * If not, default value is taken to be 0.
+ */
+
+#ifndef CONFIG_MPU_RESERVED
+#define CONFIG_MPU_RESERVED	(-1)
+#endif
+
 /* Enum for MPU region numbers */
 
 enum MPU_REG_NUM {
+	MPU_REG_NUM_RESERVE = CONFIG_MPU_RESERVED,
```

Comment:
I think we can use CONFIG_MPU_RESERVED directly in the first place of the enum. 
There is no need to introduce a new variable called MPU_REG_NUM_RESERVE